app/lib/tool/backfill/backfill_new_fields.dart

```diff
@@ -12,5 +19,72 @@ final _logger = Logger('backfill_new_fields');
 /// CHANGELOG.md must be updated with the new fields, and the next
 /// release could remove the backfill from here.
 Future<void> backfillNewFields() async {
-  _logger.info('Nothing to backfill');
+  await migrateIsBlocked();
+}
+
+/// Migrates entities from the `isBlocked` fields to the new `isModerated` instead.
+Future<void> migrateIsBlocked() async {
+  _logger.info('Migrating isBlocked...');
+  final pkgQuery = dbService.query<Package>()..filter('isBlocked =', true);
+  await for (final entity in pkgQuery.run()) {
+    await withRetryTransaction(dbService, (tx) async {
+      final pkg = await tx.lookupValue<Package>(entity.key);
+      // sanity check
+      if (!pkg.isBlocked) {
+        return;
+      }
+      pkg
+        ..isModerated = true
+        ..moderatedAt = pkg.moderatedAt ?? pkg.blocked ?? clock.now()
+        ..isBlocked = false
+        ..blocked = null
+        ..blockedReason = null;
+      tx.insert(pkg);
+    });
+
+    // sync exported API(s)
+    await apiExporter?.synchronizePackage(entity.name!, forceDelete: true);
+
+    // retract or re-populate public archive files
+    await packageBackend.tarballStorage.updatePublicArchiveBucket(
+      package: entity.name!,
+      ageCheckThreshold: Duration.zero,
+      deleteIfOlder: Duration.zero,
+    );
```

Comment:
Are we sure we want to do this? Or should we just let this happen naturally?

I don't really have any strong feelings on the subject :rofl:

Reply:
> Are we sure we want to do this? Or should we just let this happen naturally?

The tests would fail with integrity checks, otherwise we could let it happen naturally. I don't mind that it happens right away.